httpd-client/lib/projects.ts

```diff
@@ -0,0 +1,267 @@
+  path: string;
+  name: string;
+  kind: "tree" | "blob";
+}
+
+interface TreeStats {
+  commits: number;
+  branches: number;
+  contributors: number;
+}
+
+export interface Tree {
+  entries: TreeEntry[];
+  lastCommit: Commit;
+  name: string;
+  path: string;
+  stats: TreeStats;
+}
+
+export interface Remote {
+  id: string;
+  heads: Record<string, string>;
+  delegate: boolean;
+}
+
+export const projectSchema: zod.ZodSchema<Project> = zod.strictObject({
+  id: zod.string(),
+  name: zod.string(),
+  description: zod.string(),
+  defaultBranch: zod.string(),
+  head: zod.string(),
+  patches: zod.number(),
+  issues: zod.number(),
+});
+
+export const projectsSchema: zod.ZodSchema<Project[]> =
+  zod.array(projectSchema);
+
+const activitySchema: zod.Schema<Activity> = zod.strictObject({
+  activity: zod.array(zod.number()),
+});
+
+export const gitPersonSchema = zod.strictObject({
+  name: zod.string(),
+  email: zod.string(),
+});
+
+export const commitSchema = zod.strictObject({
+  id: zod.string(),
+  author: gitPersonSchema,
+  summary: zod.string(),
+  description: zod.string(),
+  committer: gitPersonSchema.merge(zod.strictObject({ time: zod.number() })),
+});
+
+export const blobSchema: zod.ZodSchema<Blob> = zod.strictObject({
+  binary: zod.boolean(),
+  content: zod.string(),
+  name: zod.string(),
+  path: zod.string(),
+  lastCommit: zod.strictObject({
+    id: zod.string(),
+    author: gitPersonSchema,
+    summary: zod.string(),
+    description: zod.string(),
+    committer: gitPersonSchema.merge(zod.strictObject({ time: zod.number() })),
+  }),
+});
+
+export const treeEntrySchema: zod.ZodSchema<TreeEntry> = zod.strictObject({
+  path: zod.string(),
+  name: zod.string(),
+  kind: zod.union([zod.literal("blob"), zod.literal("tree")]),
+});
+
+export const treeSchema: zod.ZodSchema<Tree> = zod.strictObject({
+  entries: zod.array(treeEntrySchema),
+  lastCommit: commitSchema,
+  name: zod.string(),
+  path: zod.string(),
+  stats: zod.strictObject({
+    commits: zod.number(),
+    branches: zod.number(),
+    contributors: zod.number(),
+  }),
+});
+
+export const remoteSchema: zod.ZodSchema<Remote> = zod.strictObject({
+  id: zod.string(),
+  heads: zod.record(zod.string(), zod.string()),
+  delegate: zod.boolean(),
+});
+
+export const remotesSchema: zod.ZodSchema<Remote[]> = zod.array(remoteSchema);
+
+export class Client {
+  private fetcher: Fetcher;
+
+  public constructor(fetcher: Fetcher) {
+    this.fetcher = fetcher;
+  }
+
+  public async getByDelegate(
+    delegateId: string,
+    options?: RequestOptions,
+  ): Promise<Project[]> {
+    return this.fetcher.fetchOk(
+      {
+        method: "GET",
+        path: `v1/delegates/${delegateId}/projects`,
+        options,
+      },
+      projectsSchema,
+    );
+  }
+
+  public async getById(id: string, options?: RequestOptions): Promise<Project> {
+    return this.fetcher.fetchOk(
+      {
+        method: "GET",
+        path: `v1/projects/${id}`,
```

Comment:
Wouldn't it make sense for the `v1/` part to be configured in the `fetcher`?

Reply:
Yeah, probably. Still WIP.

Reply:
Um, I think I tried. But the problem was that the root API call is without the v1. Maybe that has to change in the httpd backend?

Reply:
If we stick this endpoint: https://seed.radicle.xyz/api under https://seed.radicle.xyz/api/v1, then it would all check out.

Reply:
That's a bug yes :) cc @xphoniex

Reply:
does this https://github.com/radicle-dev/heartwood/pull/409 fix it?

Reply:
Merged.

Reply:
db896aeecb8f63c165ffa2b1fa51935815f4983f